index.js

```diff
@@ -81,7 +83,14 @@ module.exports = function (opts) {
 					deleteRequireCache(modId);
 				});
 			}
-			jasmine.addReporter(new SilentReporter(cb, errorOnFail));
+			jasmine.addReporter(new SilentReporter(function (error) {
+				if (error) {
+					cb(error);
+				} else {
+					self.emit('end');
+					cb();
```

Comment:
This is what I thought too, but apparently I was wrong. Every time I think I've understood node streams, I learn that really don't...

Apparently in gulp the end event is emitted by draining the stream with a writable stream (see https://github.com/rvagg/through2/issues/31 and https://github.com/gulpjs/gulp/issues/1637). I thought that transform streams were both readable AND writable, but I still haven't read the definitive book on streams (which I'm very interested in buying whenever one comes out), so I must be missing something.